example-app/App.tsx

```diff
@@ -40,6 +40,41 @@ const Header = () => {
 };
 
 const App = () => {
+  const [result, setResult] = React.useState<{
+    success: boolean | undefined;
+    errorReason: string | undefined;
+  }>({success: undefined, errorReason: undefined});
+
+  React.useEffect(() => {
+    Linking.addEventListener('url', event => {
+      setResult(handleDeepLink(event.url));
+    });
+
+    async function getDeepLink() {
+      let url = await Linking.getInitialURL();
+      let res = handleDeepLink(url);
+      setResult(res);
+    }
+
+    getDeepLink();
+  }, []);
+
+  const handleDeepLink = (
+    url: string | null,
+  ): {success: boolean | undefined; errorReason: string | undefined} => {
+    if (!url) {
+      return {success: undefined, errorReason: undefined};
+    }
+
+    let deeplinkUrl = new URL(url, true);
+    let error = deeplinkUrl.query.error;
+    if (error) {
+      return {success: false, errorReason: error!};
+    }
+
+    return {success: true, errorReason: undefined};
```

Comment:
```suggestion
    return {success: true};
```

If you adjust the types according to comment above, it will be invlaid to provide an error reason when success is true, and invalid to not provide an error reason when success is false.